langchain/chains/base.py

```diff
@@ -53,6 +55,16 @@ class Chain(Serializable, ABC):
             cannot return as rich of an output as `__call__`.
     """
 
+    def invoke(
+        self, input: Dict[str, Any], **kwargs: Unpack[RunnableConfig]
+    ) -> Dict[str, Any]:
+        return self(input, **kwargs)
+
+    async def ainvoke(
+        self, input: Dict[str, Any], **kwargs: Unpack[RunnableConfig]
+    ) -> Dict[str, Any]:
+        return await self.acall(input, **kwargs)
+
```

Comment:
BaseChain is getting sync and async batch methods here for free. And a basic streaming method for compatibility with other runnable objects

Reply:
+1 to the above

Reply:
How about e.g. `_invoke` to show it's private and de-emphasize it in the API ref? User should never need to call it directly

Reply:
(the user should invoke it directly if they want a single output, non-streaming, sync